File: Middle-logic/Models/SensoryStimulus.js
import { ObservableObject, ObservableObjectType } from "./ObservableObject.js";

export class SensoryStimulusType{
    static Visual = "Visual";
    static Auditory = "Auditory";
    static Olfactory = "Olfactory";
    static Kinesthetic = "Kinesthetic";
    static Taste = "Taste";
}
export class SensoryStimulus extends ObservableObject{
    constructor(id, title, description, icon, sensoryStimulusType, intensity){
        super(id, title, ObservableObjectType.SensoryStimuls, description, icon, intensity);
        this.SensoryStimulus_Type = sensoryStimulusType;
    }
    static getSensoryStimulusFromObject(obj){
        return new SensoryStimulus(
            obj.ObservableObject_ID,
            obj.Title,
            obj.Description,
            obj.Icon,
            obj.SensoryStimulus_Type,
            obj.Intensity
        );
    }
    static convertArrayToSensoryStimulusObjects(arr){
        var SensoryStimuli= arr.map(SensoryStimulus.getSensoryStimulusFromObject);
        return SensoryStimuli;
    }
}

File: OldArm_mo/Models/SensoryStimulus.cs
namespace Arm_mo.Models
{
    public class SensoryStimulus:ObservableObject
    {
        public enum SensoryStimulusType
        {
            Visual,
            Auditory,
            Olfactory,
            Kinesthetic,
            Taste
        }
        public SensoryStimulusType SensoryStimulustype { get; set; }
        protected SensoryStimulus(String title, string description,
           string icon, IntensityType intensity) : base(title, description, icon, intensity)
        {

        }
        protected SensoryStimulus(SensoryStimulusType sensoryStimulusType,String title, string description, 
            string icon, IntensityType intensity):base(title, description, icon, intensity)
        {
            SensoryStimulustype = sensoryStimulusType;

        }
        

    }
}
